packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -464,19 +465,78 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
         // TODO: This snapshot will return deltas, which we currently aren't using. We need to enable this flag to go down the "optimized"
         // snapshot code path. We should leverage the fact that these deltas are returned to speed up the deltas fetch.
-        const { headers, url } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
+        const { headers, url, isOptionsCall } = getUrlAndHeadersAndOptionsWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
```

Comment:
No need to add a new file/api. You can just check the headers to see whether it is options call or not.

Reply:
Ok, I made a change to check for headers.